### PR TITLE
Backstopjs docker

### PR DESCRIPTION
**Pull request by suryapulakhandam**
## Checklist

- [ ] SCSS
- [ ] Nunjucks macro
- [ ] A standalone example
- [ ] README/Documentation with HTML snippet
- [ ] Tested against our [testing policy](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/testing.md) (Resolution, Browser & Accessibility)
- [ ] Print stylesheet considered
- [ ] CHANGELOG entry

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +20,11 @@
   },
   "dependencies": {
     "accessible-autocomplete": "^1.6.2",
-    "sass-mq": "^4.0.2"
+    "node-sass": "^4.11.0",
```
Comment (AdamChrimes) [inline package.json]: Howcome we have installed `node-sass`, `npm` and `start` as dependencies? `dependencies` are for production dependencies and `devDependencies` are for development dependencies (so they will get installed within peoples projects when they `npm install nhsuk-frontend`

If these are required only for BackstopJS to run, can we install these only on the Docker container?

File: package.json [hunk 2]
```diff
@@ -48,6 +52,7 @@
     "htmlhint": "^0.10.1",
     "nunjucks": "^3.1.6",
     "sass-lint": "^1.12.1",
+    "start-server-and-test": "^1.7.11",
```
Comment (AdamChrimes) [inline package.json]: Where does this dependency get used? I can't see it run anywhere in the Docker image or within any `package.json` scripts.

**Comment by AdamChrimes** [comment]
@suryapulakhandam it seems the build is not passing because some of the BackstopJS tests are breaking.

File: package.json [hunk 3]
```diff
@@ -20,7 +20,11 @@
   },
   "dependencies": {
     "accessible-autocomplete": "^1.6.2",
-    "sass-mq": "^4.0.2"
+    "node-sass": "^4.11.0",
+    "npm": "^6.7.0",
```
Comment (AdamChrimes) [inline package.json]: Where does this dependency get used? I can't see it run anywhere in the Docker image or within any `package.json` scripts.

There is also a medium vulnerability within this package. https://app.snyk.io/vuln/npm:mem:20180117

File: package.json [hunk 4]
```diff
@@ -20,7 +20,11 @@
...
+    "rebuild": "^0.1.2",
```
Comment (AdamChrimes) [inline package.json]: Where does this dependency get used? I can't see it run anywhere in the Docker image or within any `package.json` scripts.

File: package.json (hunk 2 above)
Comment (suryapulakhandam) [reply package.json]: removed   "start-server-and-test": "^1.7.11",

File: package.json [hunk 5]
```diff
@@ -20,7 +20,11 @@
...
+    "sass-mq": "^4.0.2",
+    "start": "^5.1.0"
```
[the remark this answers is not in the document]
Comment (suryapulakhandam) [reply package.json]: updated the dependencies

**Comment by mikemonteith** [comment]
These 4 commits aren't very ["atomic"](https://www.freshconsulting.com/atomic-commits/). We should probably look to squash them into 1 change

**Comment by mikemonteith** [comment]
We only need docker for running backstop so maybe we should put all of the backstop-related files (.dockerignore, Dockerfile) inside the `tests/backstop/` directory

**Comment by mikemonteith** [comment]
I wonder if we can get the node app to run inside the travis running rather than inside docker. That way we could remove the slow build step caused by the Dockerfile

**Comment by suryapulakhandam** [comment]
> We only need docker for running backstop so maybe we should put all of the docker-related files (.dockerignore, Dockerfile, docker-compose.yml) inside the `tests/backstop/` directory

I have tried moving the files into backstop directory, but I keeping getting error messages. This reads as   "Can't find a suitable configuration file in this directory or any parent. Are you in the right directory?"

**Comment by AdamChrimes** [comment]
@suryapulakhandam do you need to pass the custom file path in docker-compose? something like:

`docker-compose -f path/to/docker-compose-tests.yml`

**Comment by AdamChrimes** [comment]
This has been picked up in the following PR https://github.com/nhsuk/nhsuk-frontend/pull/390

**Pull request closed without merging**